### PR TITLE
add support for dotnet build

### LINKED ISSUE #90: support dotnet build in msbuild task
[dotnet build](https://docs.microsoft.com/en-us/dotnet/articles/core/tools/dotnet-build) runs as a .net core app, as such msbuild tasks need to be compiled against netstandard to be compatible. 

msbuild tasks only targeting "original .net" will get the below error.

```
dotnet build SqlPersistence.sln
  SharedMessages -> C:\Code\docs.particular.net\samples\sql-persistence\simple\SqlPersistence_2\SharedMessages\bin\Debug\SharedMessages.dll
C:\Code\docs.particular.net\packages\NServiceBus.Persistence.Sql.MsBuild.2.1.0\build\NServiceBus.Persistence.Sql.MsBuild.targets(14,5): 
error MSB4062: The "NServiceBus.Persistence.Sql.ScriptBuilderTask" task could not be loaded from the assembly C:\Code\docs.particular.net\packages\NServiceBus.Persistence.Sql.MsBuild.2.1.0\build\..\NServiceBus.Persistence.Sql.ScriptBuilderTask.dll. 
Could not load file or assembly 'Microsoft.Build.Utilities.v4.0, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
 The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [C:\Code\docs.particular.net\samples\sql-persistence\simple\SqlPersistence_2\ServerShared\ServerShared.csproj]
  Client -> C:\Code\docs.particular.net\samples\sql-persistence\simple\SqlPersistence_2\Client\bin\Debug\Client.exe

Build FAILED.

C:\Code\docs.particular.net\packages\NServiceBus.Persistence.Sql.MsBuild.2.1.0\build\NServiceBus.Persistence.Sql.MsBuild.targets(14,5): 
error MSB4062: The "NServiceBus.Persistence.Sql.ScriptBuilderTask" task could not be loaded from the assembly C:\Code\docs.particular.net\packages\NServiceBus.Persistence.Sql.MsBuild.2.1.0\build\..\NServiceBus.Persistence.Sql.ScriptBuilderTask.dll. 
Could not load file or assembly 'Microsoft.Build.Utilities.v4.0, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. 
The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [C:\Code\docs.particular.net\samples\sql-persistence\simple\SqlPersistence_2\ServerShared\ServerShared.csproj]
```

### PR DESCRIPTION
**Pull request by SimonCropp**
fixes #90

### CODE REVIEW

File: packaging/nuget/NServiceBus.Persistence.Sql.MsBuild.targets [hunk 1]
```diff
@@ -1,12 +1,20 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <PropertyGroup>
-    <ScriptDirectory>$(ProjectDir)$(IntermediateOutputPath)NServiceBus.Persistence.Sql\</ScriptDirectory>
```
Comment (SimonCropp) [inline packaging/nuget/NServiceBus.Persistence.Sql.MsBuild.targets]: @MarcinHoppe `$(ProjectDir)$(IntermediateOutputPath)` dont exist until later in the build. i needed to move them down to get it working in the new project system

Comment (MarcinHoppe) [inline packaging/nuget/NServiceBus.Persistence.Sql.MsBuild.targets]: So it does not work now (after move of the persistence itself to VS 2017) but it did work before?

Comment (SimonCropp) [inline packaging/nuget/NServiceBus.Persistence.Sql.MsBuild.targets]: after moving to the new vs2017 csproj system it did not correctly evaluate that property

File: src/Integration/OracleSample/Program.cs [hunk 2]
```diff
@@ -13,11 +13,12 @@ static void Main()
...
-        {
-            persistence.SqlVariant(SqlVariant.MySql);
-            persistence.TablePrefix("MySql");
-            persistence.ConnectionBuilder(() => new OracleConnection(connection));
-        });
+        return EndpointStarter.Start("SqlPersistence.Sample.Oracle",
+            persistence =>
+            {
+                persistence.SqlVariant(SqlVariant.MySql);
+                persistence.TablePrefix("MySql");
```
Comment (weralabaj) [inline src/Integration/OracleSample/Program.cs]: seems there's some mix-up here, oracle endpoint but MySql variant and MySql prefix?

Comment (SimonCropp) [inline src/Integration/OracleSample/Program.cs]: that was not me https://github.com/Particular/NServiceBus.Persistence.Sql/blob/develop/src/Integration/OracleSample/Program.cs

Comment (weralabaj) [inline src/Integration/OracleSample/Program.cs]: I know, but I see it now ;)

Comment (SimonCropp) [inline src/Integration/OracleSample/Program.cs]: yeah i will fix on this PR

**Comment by weralabaj** [comment]
@MarcinHoppe can you have a look too? I've reviewed new changes and seems to be working fine locally (although I don't have Oracle configured, so this part is unchecked ;)).

**Comment by MarcinHoppe** [comment]
@weralabaj It builds fine and ATTs pass on Oracle.

**Pull request merged**